Parse/Infrastructure/Control/ParseAddOperation.cs

```diff
@@ -7,31 +7,118 @@
+        }
+
+        var result = Conversion.To<IList<object>>(oldValue).ToList();
+        foreach (var obj in Data)
+        {
+            if (!result.Contains(obj)) // Ensure no duplicates
+            {
+                result.Add(obj);
+            }
+        }
+        return result;
+    }
+
+    public IDictionary<string, object> ConvertToJSON(IServiceHub serviceHub = default)
+    {
+        // Convert the data into JSON-compatible structures
+        var encodedObjects = Data.Select(EncodeForParse).ToList();
 
-        public object Encode(IServiceHub serviceHub) => new Dictionary<string, object>
+        return new Dictionary<string, object>
         {
             ["__op"] = "Add",
-            ["objects"] = PointerOrLocalIdEncoder.Instance.Encode(Data, serviceHub)
+            ["objects"] = encodedObjects
         };
+    }
 
-        public IParseFieldOperation MergeWithPrevious(IParseFieldOperation previous) => previous switch
+    private object EncodeForParse(object obj)
+    {
+        return obj switch
         {
-            null => this,
-            ParseDeleteOperation { } => new ParseSetOperation(Data.ToList()),
-            ParseSetOperation { } setOp => new ParseSetOperation(Conversion.To<IList<object>>(setOp.Value).Concat(Data).ToList()),
-            ParseAddOperation { } addition => new ParseAddOperation(addition.Objects.Concat(Data)),
-            _ => throw new InvalidOperationException("Operation is invalid after previous operation.")
-        };
+            // Handle pointers
+            ParseObject parseObj => new Dictionary<string, object>
+            {
+                ["__type"] = "Pointer",
+                ["className"] = parseObj.ClassName,
+                ["objectId"] = parseObj.ObjectId
+            },
+
+            // Handle GeoPoints
+            ParseGeoPoint geoPoint => new Dictionary<string, object>
+            {
+                ["__type"] = "GeoPoint",
+                ["latitude"] = geoPoint.Latitude,
+                ["longitude"] = geoPoint.Longitude
+            },
+
+            // Handle Files
+            ParseFile file => new Dictionary<string, object>
+            {
+                ["__type"] = "File",
+                ["name"] = file.Name,
+                ["url"] = file.Url
+            },
 
-        public object Apply(object oldValue, string key) => oldValue is { } ? Conversion.To<IList<object>>(oldValue).Concat(Data).ToList() : Data.ToList();
+            // Handle Relations
+            ParseRelationBase relation => new Dictionary<string, object>
+            {
+                ["__type"] = "Relation",
+                ["className"] = relation.TargetClassName
+            },
 
-        public IEnumerable<object> Objects => Data;
+            // Handle primitive types
+            string or int or long or float or double or decimal or bool => obj,
+
+            // Handle Bytes
+            byte[] bytes => new Dictionary<string, object>
```

Comment:
Should this be separated out/reused

Reply:
I would see why!
But did not find the functional need to reorganize.
If commenting out works for now, I can do that